websites/N/namu.wiki/presence.ts

```diff
@@ -0,0 +1,189 @@
+  "mypage": "My Page",
+  "change_email": "Change Email",
+  "change_password": "Change Password",
+  "activate_otp": "Activate OTP"
+},
+// /RecentChanges?logtype=(search)
+changesMapping: interfaceMapping = {
+  "all": "All Changes",
+  "create": "Created Documents",
+  "delete": "Deleted Documents",
+  "move": "Moved Documents",
+  "revert": "Reverted Documents"
+},
+// /RecentDiscuss?logtype=(search)
+discussMapping: interfaceMapping = {
+  "normal_thread": "Normal Thread",
+  "old_thread": "Old Thread",
+  "closed_thread": "Closed Thread",
+  "open_editrequest": "Opened Edit Request",
+  "accepted_editrequest": "Accepted Edit Request",
+  "closed_editrequest": "Closed Edit Request",
+  "old_editrequest": "Old Edit Request"
+},
+// /contribution/(type)/(username)/(contributeType)
+validateContributeUrl = /\/contribution\/(.+)\/(.+)\/(.+)/;
+
+let currentPage = document.location.pathname, currentTime = Date.now();
+presence.on("UpdateData", async () => {
+  const privacy = await presence.getSetting("privacy"),
+  showTimestamp = await presence.getSetting("show_timestamp"),
+  presenceData: PresenceData = {
+    largeImageKey: "namu"
+  };
+
+  if (document.location.hostname !== "namu.wiki") return;
+  const path = document.location.pathname,
+        params = document.location.search,
+        parsedUrl = path.split("/"); // It's a very bad design, but they have a slash document.
+  if (parsedUrl === null) return;
+
+  /**
+   * 
+   * Setting Details & State
+   * 
+   */
+  let privateMode = false;
+  const action = parsedUrl[1], details = boardTypeMapping[action];
+  presenceData.details = ((details === undefined) ? "Unknown Action" : details);
+
+  let page;
+  /* View Contribute */
+  if ((page = validateContributeUrl.exec(path))) {
+    if (page[1] === "author") {
+      page = `User: ${page[2]}`;
+    } else {
+      privateMode = true;
+      page = "IP User";
+    }
+
+  /* View Membership */
+  } else if ((page = validateMembershipUrl.exec(path))) {
+    privateMode = true; // Privacy
+    presenceData.details = "Member Page";
+    page = membersMapping[page[1]];
+
+  /* Searching */
+  // /Search?q=(search)
+  } else if (action === "Search") {
+    page = getParam(params, 'q');
+    if (page.length === 0) page = "Blank Query...";
+
+  /* View Recent Discuss History */
+  } else if (action === "RecentDiscuss")
+    page = discussMapping[getParam(params, 'logtype')];
+
+  /* Recent Changes History */
+  else if (action === "RecentChanges")
+    page = changesMapping[getParam(params, 'logtype')];
+
+  /* View Discuss Thread */
+  else if (action === "thread" || action === "edit_request")
+    page = document.querySelector(`h1 > a`).textContent; // H1 Tag Only one
+
+  /* Other */
+  else if (details !== undefined) page = decodeURI(path.substring(`/${action}/`.length));
+  else page = null;
+
+  if (page === null || page === undefined) page = "";
+  presenceData.state = ((page.length > 128) ? `${page.slice(0, 120)}...` : page);
```

Comment:
don't set blank or null variables to presenceData

Reply:
Is there anything else I can use except blank when there is no information to display?

Reply:
Check your if above it and just modify it.